lib/server.js

```diff
@@ -1405,7 +1405,16 @@ Server.prototype._routeErrorResponse = function _routeErrorResponse(
         self.handleUncaughtExceptions &&
         self.listenerCount('uncaughtException') > 1
     ) {
-        self.emit('uncaughtException', req, res, req.route, err);
+        self.emit(
+            'uncaughtException',
+            req,
+            res,
+            req.route,
+            err,
+            function uncaughtExceptionCompleted() {
```

Comment:
Please put a comment about the why here and describe what will happen with and without domains.

Reply:
+1

Reply:
@hekike @rajatkumar Added some comments here, let me know if that addresses your feedback!